pkg/agent/nodeportlocal/k8s/controller.go

```diff
@@ -112,12 +175,136 @@ func (c *Controller) EnqueueObj(obj interface{}) {
 	c.queue.Add(podKey)
 }
 
-func (c *Controller) Worker() {
+func (c *NPLController) checkDeletedSvc(obj interface{}) (*corev1.Service, error) {
+	deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+	if !ok {
+		return nil, fmt.Errorf("Received unexpected object: %v", obj)
+	}
+	svc, ok := deletedState.Obj.(*corev1.Service)
+	if !ok {
+		return nil, fmt.Errorf("DeletedFinalStateUnknown object is not of type Service: %v", deletedState.Obj)
+	}
+	return svc, nil
+}
+
+func (c *NPLController) enqueueSvcUpdate(oldObj, newObj interface{}) {
+	// In case where the app selector in Service gets updated from one valid selector to another
+	// both sets of Pods (corresponding to old and new selector) need to be considered.
+	newSvc := newObj.(*corev1.Service)
+	oldSvc := oldObj.(*corev1.Service)
+
+	// Return if Service ResourceVersions don't change.
+	if oldSvc.ResourceVersion == newSvc.ResourceVersion {
+		return
+	}
```

Comment:
why the inconsistency between Pods and Services? For Pods, you perform that check before calling the enqueue function

Reply:
This is because there is no separate enqueue function for Pod updates as compared to Service updates (enqueueSvcUpdate). There is a separate function for Service updates because we queue pods from both old and new services. For Pod updates we eventually just enqueue the new Pods.

Reply:
thanks for the explanation